### PR TITLE
mesh_sampling tool: Add support for colors

### PR DESCRIPTION
**Pull request by mintar**

### CODE REVIEW

File: tools/mesh_sampling.cpp [hunk 1]
```diff
@@ -81,7 +79,18 @@ randomPointTriangle (float a1, float a2, float a3, float b1, float b2, float b3,
 }
 
 inline void
-randPSurface (vtkPolyData * polydata, std::vector<double> * cumulativeAreas, double totalArea, Eigen::Vector4f& p, bool calcNormal, Eigen::Vector3f& n)
+randomPointTriangle (float a1, float a2, float a3, float b1, float b2, float b3, float c1, float c2, float c3,
```
Comment (taketwo) [inline tools/mesh_sampling.cpp]: Why do we need this overload? (see next comment)

File: tools/mesh_sampling.cpp [hunk 2]
```diff
@@ -135,7 +169,8 @@ uniform_sampling (vtkSmartPointer<vtkPolyData> polydata, size_t n_samples, bool
   {
     Eigen::Vector4f p;
     Eigen::Vector3f n;
-    randPSurface (polydata, &cumulativeAreas, totalArea, p, calc_normal, n);
+    Eigen::Vector3i c;
```
Comment (taketwo) [inline tools/mesh_sampling.cpp]: This can as well be `Eigen::Vector4f c`, which can be passed directly to the existing version of `randomPointTriangle()`.

**Comment by mintar** [comment]
@taketwo : I've updated the PR according to your comments.

**Comment by taketwo** [comment]
Thanks!

**Pull request merged**